public/discussion.php

```diff
@@ -47,112 +50,113 @@
+                    <ul>
+                    <?php foreach ($poll->poll_options as $option) { ?>
+                    <li>
+                        <?php echo sprintf("%s - %d votes - %.2f%%", $option->content, $poll_result->votes[$option->id][0], $poll_result->votes[$option->id][1] * 100) ?></li>
+                    </li>
+                    <?php } ?>
+                    </ul>
+                    <?php
+            } else { ?>
                 <?php
-                    if ($current_user_may_comment) {
-                        $user_id = $_SESSION["current_user"]->id;
-                        $commentable_id = $discussion_message->id;
-                        $commentable_type = "DiscussionMessage";
-                        if (get_current_role() == "instructor" || get_current_role() == "ta") {
-                            include "new_comment_form.php";
-                        }
-                    }
-                ?>
-                </td>
-                
-                </td>
-                <!-- Reply button -->
-                <td><a href="discussions.php" class="replyMessage">Reply</a></td>
-                <!-- Reply form -->
-                <td>
-                    <form method="post" action="discussion.php" class="replyMessageForm" style="display: none;">
-                        <label for="content">Content</label>
-                        <input type="text" name="content" id="content">
-                        <input type="hidden" id="discussion_id" name="discussion_id" value="<?php echo $discussion->id; ?>">
-                        <input type="hidden" id="user_id" name="user_id" value="<?php echo $_SESSION["current_user"]->id; ?>">
-                        <!-- Put in actual ID of post to reply to-->
-                        <input type="hidden" id="replies_to" name="replies_to" value="<?php echo $discussion_message->id ?>">
-                        <input type="submit" name="submit" value="Submit">
+                if (get_current_role() == "student") {?>
+                    <form method="post" action="poll.php" id="pollVote">
+                        <input type="hidden" id="poll_id" name="poll_id" value="<?php echo $poll->id; ?>">
+                        <p>Options:</p>
+                        <?php foreach ($poll->poll_options as $option) { ?>
+                            <input type="radio" id="vote_option_<?php echo $option->id ?>" name="vote_option" value="<?php echo $option->id ?>">
+                            <label for="vote_option_<?php echo $option->id ?>"><?php echo $option->content ?></label><br>
+                        <?php }?>
+                        <input type="submit" name="submit" value="Vote">
                     </form>
-                </td>
-            </tr>
-        <?php } ?>
+                    <?php
+                } ?>
+            <?php }
+            }?>
+            <?php
+            // TODO: Only display this to TA's of this course!
+            if (true) {
```

Comment:
This is the code from the old version, the new version checks if the user has the permission to comment